cmd/minishift/cmd/ssh.go

```diff
@@ -34,9 +38,13 @@ var sshCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 		defer api.Close()
+
 		err := cluster.CreateSSHShell(api, args)
 		if err != nil {
-			atexit.ExitWithMessage(1, fmt.Sprintf("Cannot establish SSH connection to the VM: %s", err.Error()))
+			if err.Error() == SshCommunicationError {
```

Comment:
I had a feeling that return code 255 is not enough for finding out error SSH connection. But did some testing around it and @gbraad pointed me to https://man.openbsd.org/ssh#EXIT_STATUS . Usecases where VM's SSH service is down or VM is shutdown the error is proper (The error when VM is shutdown is coming from cluster.go). Looks good to me.